### DISCUSSION TITLE
[Suggestion] /cmi removeitem

### DISCUSSION THREAD

**Discussion by Segysmundo** [Zrips/CMI#1349] [Feature Requests]
What do you think of adding a command like:
**/cmi removeitem [player] [item] (amount) (name) (lore) (enchants)**
or
**/cmi takeitem [player] [item] (amount) (name) (lore) (enchants)**

If the player has an item matching the arguments of the command it will delete it from his inventory and if it doesn't match it will just say the item wasn't found.

Maybe make it so you can specify if name or lore has a certain color, for ex:

/cmi removeitem **Joe** item:**stick** amount:**4** name:**&6SuperStick** lore:**&4A nice stick** enchants:**Sharpness 1**

I'm sorry if something like this already exists and I'm pointlessly suggesting but I couldn't find it.

**Comment by mrfloris** [comment]
Using customalias you can check against  ```item:[item:(data)(-amount)]#?!``` which will let a- not run the command if it isn't found, b- tell them they're missing the required item, and c- take it if it can. 

We use this for trading with cmi ic and cmi aliaseditor and attachcommand.

**Comment by Segysmundo** [comment]
> Using customalias you can check

You mean using specialized commands?

**Edit:** And does this requirement check name and lore or is it only for item and data?

**Comment by Zrips** [comment]
This is some what same feature as /cmi search which can remove items by specified criteria from players inventory and ender chest.

**Comment by Segysmundo** [comment]
Yep, I just thought, this way it could be added to our server moderators without having to give them the **/cmi search** command, to automated systems (menu commands, server events) and some other things. And is faster and less resource intensive than **/cmi search**
